### DISCUSSION TITLE
microVM and single tenancy

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [firecracker-microvm/firecracker#4485] [Q&A]
Could a microVM be a single dedicated server ie. one organization with one (physical) server with one application? Further, could Firecracker manage application software updates to the microVM's so, like with multi-tenancy, there is only one code base? Thanks

**Comment by zulinx86** [comment]
Hello @dineshbvadhia ,
Thank you for your question!

Firecracker itself provides the ability to boot a virtual machine and it allows you to boot multiple virtual machines on a single physical server by creating multiple Firecracker processes. How to use microVMs is up to you: what to run inside a microVM, how many applications to run inside a microVM, how to assign microVMs to someone (people, orgs, etc.), how many microVMs to boot on a single (physical) host, etc. That also means application-level mechanisms (like updating application softwares running inside multiple microVMs) need to be built up on users’ side.

Thanks,
